### ISSUE TITLE
Feature request to the Extension

### ISSUE THREAD

**Issue by deepraj02** [RobertBrunhage/flutter-riverpod-snippets#36]
Hey Robert, I've been using your extension for a very long time and I love it.
However I've some cool ideas in my mind regarding this Extension:
 - Code actions (like the ones Provided by the BLOC extension)
<img width="387" alt="image" src="https://github.com/RobertBrunhage/flutter-riverpod-snippets/assets/63138398/8940fff0-2294-460b-91ab-ecfefc6a3077">

- [ ] Convert to StatefulConsumer, Consumer, HooksConsumer (from stateless and Stateful widgets)
- [ ] Wrap with ConsumerWidget 

These are some small changes that I feel will be good features for upcoming release.
And if you allow me I'd like to add these features to the Extension myself.

**Comment by yang-lile** [comment]
[riverpod_lint](https://pub.dev/packages/riverpod_lint)
This package can solve these issue.

**Comment by minusgat** [comment]
@RobertBrunhage your extension is incredible and very useful, I greatly agree with @@deepraj02. I came to this Issue looking for the same functionality and I also think it would add a lot of value to your extension. Is there any way to contribute to this?

**Comment by Trung15010802** [comment]
riverpod_lint isn't stable. I still want this extension will add this feature. 
Ping @RobertBrunhage

**Comment by deepraj02** [comment]
Yeah I feel it is  better to include the snippets in the extension rathen then depending on another dependency/package.

**Comment by deepraj02** [comment]
The project seems to be in a dead state how about I create a fork and add the requested features (as a seperate extension)

**Comment by pythonhubdev** [comment]
Guess I will fork it and make contributions as needed. I think @RobertBrunhage might be busy on maintaining this. I will try to release a new version soon.

**Comment by deepraj02** [comment]
@pythonhubdev I'm also working on the same, can we collaborate?

**Comment by UsamaKarim** [comment]
It's an excellent idea, It will make life easy.

**Comment by Trung15010802** [comment]
Where can i see your process?

**Comment by Trung15010802** [comment]
**I wrote an extension here:**
https://marketplace.visualstudio.com/items?itemName=QuangTrung.riverpod-widget-refactor

**Comment by quyenvsp** [comment]
Hi, since it already supported by Official riverpod_lint, then if you guys plan to add to this plugin please make sure have option to disable it. I do not want my context menu have duplicate item, thank you so much!

**Comment by RobertBrunhage** [comment]
Hello everyone!

Sorry I haven't really been active with this extension as I don't personally use it anymore (not using riverpod).

The core goal of this was providing snippets only, so that it could be reused in other environments as well if needed (such as neovim).

For that reason I am closing this issue.
